app/frontend/src/features/auth/login/Login.tsx

```diff
@@ -20,6 +20,7 @@ export default function Login() {
   const error = useTypedSelector((state) => state.auth.error);
 
   const location = useLocation();
+  const { from } = (location.state as any) || { from: { pathname: "/" } };
```

Comment:
I was thinking about going straight to the needed value and not creating an object.
```suggestion
  const location = useLocation<undefined | { from: Location }>();
  const redirectTo = location.state?.from.pathname || "/";
```

And then also change the redirect.to

Reply:
Trying to understand what is happening here. `state?.` tells typescript to stop checking the rest. This `<undefined | { from: Location }>` means the value returned can be undefined or an object. But can you explain the `{ from: Location }` part?

Reply:
`useLocation<undefined | { from: Location }>` means that the state property of the returned location is either undefined or an object with a from property containing a value of type Location.